src/cli/command-processor.ts

```diff
@@ -0,0 +1,225 @@
+import {Worker} from 'worker_threads';
+import yargs from 'yargs/yargs';
+import * as readline from 'readline';
+
+/**
+ * A flag for the worker thread to indicate it's ready to recieve messages.
+ */
+export const READY_FLAG = 'READY';
+
+/**
+ * An interface to formalize the message being sent to the worker thread.
+ */
+export interface IntentMessage {
+  intentType: string;
+  requestId: string;
+}
+
+/**
+ * A class containing all parameters needed to process an Identify command.
+ */
+export class IdentifyMessage implements IntentMessage {
```

Comment:
could we use the existing local home sdk types?

Reply:
These are for formatting messages to pass to the worker, so they're different from the ones in the Local Home SDK.